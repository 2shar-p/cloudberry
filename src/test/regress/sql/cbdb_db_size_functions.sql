DROP TABLE IF EXISTS cbdbheapsizetest;
DROP TABLE IF EXISTS cbdbaosizetest;
DROP TABLE IF EXISTS t_ext;
-- create heap table
CREATE TABLE cbdbheapsizetest(a int);
INSERT INTO cbdbheapsizetest select generate_series(1, 1000);

-- create ao table
CREATE TABLE cbdbaosizetest (a int) WITH (appendonly=true, orientation=row);
insert into cbdbaosizetest select generate_series(1, 100000);

-- create EXTERNAL table
CREATE EXTERNAL TABLE t_ext (a integer) LOCATION ('file://127.0.0.1/tmp/foo') FORMAT 'text';

WITH cbdbrelsize AS (
	SELECT *
	FROM cbdb_relation_size((SELECT array_agg(oid) FROM pg_class WHERE relkind != 'f'))
	), pgrelsize AS (
	SELECT pg_relation_size(oid) as size, relname, oid FROM pg_class
	WHERE relkind != 'f'
)
SELECT pgrelsize.relname, pgrelsize.size, cbdbrelsize.size
FROM pgrelsize FULL JOIN cbdbrelsize
ON pgrelsize.oid = cbdbrelsize.reloid
WHERE pgrelsize.size != cbdbrelsize.size;

SELECT size FROM cbdb_relation_size(array['t_ext'::regclass]);
SELECT * FROM cbdb_relation_size(array[]::oid[], 'main');
SELECT size FROM cbdb_relation_size(array['cbdbheapsizetest'::regclass], 'fsm');
