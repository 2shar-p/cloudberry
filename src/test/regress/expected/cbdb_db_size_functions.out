DROP TABLE IF EXISTS cbdbheapsizetest;
NOTICE:  table "cbdbheapsizetest" does not exist, skipping
DROP TABLE IF EXISTS cbdbaosizetest;
NOTICE:  table "cbdbaosizetest" does not exist, skipping
DROP TABLE IF EXISTS t_ext;
NOTICE:  table "t_ext" does not exist, skipping
-- create heap table
CREATE TABLE cbdbheapsizetest(a int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO cbdbheapsizetest select generate_series(1, 1000);
-- create ao table
CREATE TABLE cbdbaosizetest (a int) WITH (appendonly=true, orientation=row);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into cbdbaosizetest select generate_series(1, 100000);
-- create EXTERNAL table
CREATE EXTERNAL TABLE t_ext (a integer) LOCATION ('file://127.0.0.1/tmp/foo') FORMAT 'text';
WITH cbdbrelsize AS (
	SELECT *
	FROM cbdb_relation_size((SELECT array_agg(oid) FROM pg_class WHERE relkind != 'f'))
	), pgrelsize AS (
	SELECT pg_relation_size(oid) as size, relname, oid FROM pg_class
	WHERE relkind != 'f'
)
SELECT pgrelsize.relname, pgrelsize.size, cbdbrelsize.size
FROM pgrelsize FULL JOIN cbdbrelsize
ON pgrelsize.oid = cbdbrelsize.reloid
WHERE pgrelsize.size != cbdbrelsize.size;
 relname | size | size 
---------+------+------
(0 rows)

SELECT size FROM cbdb_relation_size(array['t_ext'::regclass]);
WARNING:  skipping "t_ext" --- cannot calculate this foreign table size
 size 
------
    0
(1 row)

SELECT * FROM cbdb_relation_size(array[]::oid[], 'main');
 reloid | size 
--------+------
(0 rows)

SELECT size FROM cbdb_relation_size(array['cbdbheapsizetest'::regclass], 'fsm');
 size 
------
    0
(1 row)

